AppCenterDistribute/AppCenterDistribute/MSDistribute.m

```diff
@@ -736,9 +736,12 @@ - (BOOL)checkForUpdatesAllowed {
   // Check if we are not in AppStore or TestFlight environments.
   BOOL environmentOkay = [MSUtility currentAppEnvironment] == MSEnvironmentOther;
 
+  // Check if we are currently in guided access mode
+  BOOL noGuidedAccessMode = !UIAccessibilityIsGuidedAccessEnabled();
```

Comment:
We need a unit test for this one. 😄 
Also, we should add this in log message so that users can understand why Distribute doesn't work in this case.
The log message is [here](https://github.com/Microsoft/AppCenter-SDK-Apple/blob/f38bab88d14dd80dabc09d3d7c457eed48c47641/AppCenterDistribute/AppCenterDistribute/MSDistribute.m#L311)